src/decisionengine/framework/engine/DecisionEngine.py

```diff
@@ -348,14 +389,19 @@ def start_channels(self):
         else:
             self.logger.debug(f"Found channels: {self.channel_config_loader.get_channels().items()}")
 
-        for name, config in self.channel_config_loader.get_channels().items():
+        # We start channels in parallel to avoid unnecessary synchronization between channels.
+        exec_results = {}
+        channel_configs = self.channel_config_loader.get_channels()
+        num_thread_workers = max(1, len(channel_configs))  # Must have at least one worker
```

Comment:
Can we add a debug for how many workers we calculated, might be handy for really large configs....